### ISSUE TITLE
App crashing when moved from background to foreground

### ISSUE THREAD

**Issue by AndreiTaranovschi** [react-navigation/react-navigation#10146]
### Current behavior

Current Behavior Use react navigation linking props for deeplinking in My app. Not used useLinking any part of the code and also have declared 'android:launchMode=singleTask' in manifest file Getting below crash sometimes when app is moved from background to foreground

Fatal Exception: com.facebook.react.common.JavascriptException: Error: Looks like you have configured linking in multiple places. This is likely an error since deep links should only be handled in one place to avoid conflicts. Make sure that:
- You are not using both 'linking' prop and 'useLinking'
- You don't have 'useLinking' in multiple components
- You have set 'android:launchMode=singleTask' in the '<activity />' section of the 'AndroidManifest.xml' file to avoid launching multiple instances

This error is located at:
    in NavigationContainer
    in RCTView
    in Unknown
    in SafeAreaView
    in Unknown
    in Connect(Component)
    in PersistGate
    in Provider
    in Unknown
    in RootApp
    in CodePushComponent
    in RCTView
    in Unknown
    in RCTView
    in Unknown
    in AppContainer, js engine: hermes, stack:
anonymous@1:1161969
flushPassiveEffectsImpl@1:178330
anonymous@1:296362
runWithPriority@1:141995
flushPassiveEffects@1:177886
anonymous@1:177830
V@1:295254
w@1:295578
anonymous@1:210195
_callTimer@1:209434
callTimers@1:211271
__callFunction@1:118825
anonymous@1:117312
__guard@1:118270
callFunctionReturnFlushedQueue@1:117270

       at com.facebook.react.modules.core.ExceptionsManagerModule.reportException(ExceptionsManagerModule.java:83)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
       at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:151)
       at com.facebook.react.bridge.queue.NativeRunnable.run(NativeRunnable.java)
       at android.os.Handler.handleCallback(Handler.java:790)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27)
       at android.os.Looper.loop(Looper.java:164)
       at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:226)
       at java.lang.Thread.run(Thread.java:764)

### Expected behavior

Should not crash

### Reproduction

It is an error from firebase idk

### Platform

- [X] Android
- [X] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [X] @react-navigation/bottom-tabs
- [X] @react-navigation/drawer
- [ ] @react-navigation/material-bottom-tabs
- [ ] @react-navigation/material-top-tabs
- [ ] @react-navigation/stack
- [ ] @react-navigation/native-stack

### Environment

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               | 5.9.8
| @react-navigation/bottom-tabs          |  5.11.15
| @react-navigation/stack                 | 5.14.9
| react-native-safe-area-context          | 3.3.2
| react-native-reanimated                | 2.2.3
| react-native-tab-view                  | 3.1.1
| react-native-pager-view                | 5.4.7
| react-native                           | 0.65.1

**Comment by cjhines** [comment]
Also encountering this on Android 

| package                                | version |
| -------------------------------------- | ------- |
| react-native                           |    0.65.1     |
| @react-navigation/native               |     6.0.2    |
| @react-navigation/native-stack         |     6.0.2    |
| react-native-screens                   |    3.6.0     |
| react-native-safe-area-context         |    3.3.2     |
| react-native-gesture-handler           |    1.10.3     |
| react-native-reanimated                |     2.2.2    |
| react-native-unimodules                                   |    0.14.9     |

**Comment by cjhines** [comment]
> Couldn't find version numbers for the following packages in the issue:
> 
> * `@react-navigation/drawer`
> * `@react-navigation/material-bottom-tabs`
> * `@react-navigation/material-top-tabs`
> 
> Can you update the issue to include version numbers for those packages? The version numbers must match the format 1.2.3.
> 
> The versions mentioned in the issue for the following packages differ from the latest versions on npm:
> 
> * `@react-navigation/native` (found: `5.9.8`, latest: `6.0.6`)
> * `@react-navigation/bottom-tabs` (found: `5.11.15`, latest: `6.0.9`)
> * `@react-navigation/stack` (found: `5.14.9`, latest: `6.0.11`)
> 
> Can you verify that the issue still exists after upgrading to the latest versions of these packages?

I just verified this crash to occur on latest `@react-navigation` versions:

| package                                | version |
| -------------------------------------- | ------- |
| react-native                           |    0.65.1     |
| @react-navigation/native               |     6.0.6    |
| @react-navigation/native-stack         |     6.0.11    |
| react-native-screens                   |    3.9.0     |
| react-native-safe-area-context         |    3.3.2     |
| react-native-gesture-handler           |    1.10.3     |
| react-native-reanimated                |     2.2.2    |
| react-native-unimodules                                   |    0.14.9     |

Reproduction:
1. Put app in background
2. Open many memory-intensive app so OS kills app for memory saving.
3. Reopen app.
4. Crashes to homescreen.

**Comment by cjhines** [comment]
Related: https://github.com/software-mansion/react-native-screens/issues/1222

**Comment by UnySonic** [comment]
How is it related with my issue?

**Comment by jkadamczyk** [comment]
I noticed the same, it happens only on Android only.
I also noticed it happens only for a couple of users – around 40 and usually happens more than once for these users. Not sure if it is related, but I have some conditional logic, that shows the navigation container or something else in some cases. The conditional logic is activated pretty rarely, like when the device is rooted or the fonts are not yet loaded (some expo snippet).

Is it possible that it occurs because Navigation Container is mounted more than once?
